dev_guide/builds.adoc

```diff
@@ -846,20 +846,28 @@ source:
 
 == Binary Source
 
-When the `*BuildConfig.spec.source.type*` is `*Binary*`, the build will expect a
-binary as input, and an inline Dockerfile is optional.
-
-The binary is generally assumed to be a tar, gzipped tar, or zip file depending
-on the strategy. For `*Docker*` builds, this is the build context and an
-optional Dockerfile may be specified to override any Dockerfile in the build
-context. For `*Source*` builds, this is assumed to be an archive as described
-above. For `*Source*` and `*Docker*` builds, if `*binary.asFile*` is set the
-build context will consist of a single file named by the value of `*binary.asFile*`. 
-The `*contextDir*` field may be used when an archive is provided. Custom builds will 
-receive this binary as input on standard input (`stdin`).
-
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
+When the `*BuildConfig.spec.source.type*` is `*Binary*`, the source content will be sent
+as input to the build via a binary stream and  `oc start-build` should be executed with either
+the `--from-dir`, `--from-file`, or `--from-repo` options.
+
+The binary stream will be processed in one of two ways:
+
+1) If `*binary.asFile*` is set, then the bytes from the binary stream will be considered a single file.  The binary
+stream will be stored in a file on the builder's file system with the name as specified as `*binary.asFile*` for the builder to use. 
```

Comment:
Yeah, in fact, the documentation also has a blurb that "Binary and Git are mutually exclusive source types" ... from a BuildConfig perspective, perhaps, but with the `--from*` options, that does not seem entirely true, at least from a broader perspective.  Note the extended tests at https://github.com/openshift/origin/blob/master/test/extended/builds/start.go#L85-L146